java/ql/lib/semmle/code/java/environment/SystemProperty.qll

```diff
@@ -0,0 +1,230 @@
+import java
+private import semmle.code.java.frameworks.apache.Lang
+
+/**
+ * Gets Expr that return the value of `propertyName` from `System.getProperty()`.
+ */
+Expr getSystemProperty(string propertyName) {
+  result = getSystemPropertyFromSystem(propertyName) or
+  result = getSystemPropertyFromFile(propertyName) or
+  result = getSystemPropertyFromApacheSystemUtils(propertyName) or
+  result = getSystemPropertyFromApacheFileUtils(propertyName) or
+  result = getSystemPropertyFromGuava(propertyName) or
+  result = getSystemPropertyFromOperatingSystemMXBean(propertyName) or
+  result = getSystemPropertyFromSpringProperties(propertyName)
+}
```

Comment:
This API has been kicking around my head for the past few weeks. Trying to unify all the system property accessor `FieldAccess` & `MethodAccess` into a common API has been something I've been trying to think through how to do well.

I don't think this could be done with a class (given that it needs to take an argument), but if there's a better way to do this, I'm happy to hear it.